packages/core/src/controllers/relayer.ts

```diff
@@ -141,6 +142,12 @@ export class Relayer extends IRelayer {
     this.relayUrl = relayUrl || this.relayUrl;
     this.transportExplicitlyClosed = false;
     await this.provider.connect();
+    // wait the subscriber to finish resubscribing to its topics
+    await new Promise<void>((resolve) => {
+      this.subscriber.on(SUBSCRIBER_EVENTS.resubscribed, () => {
```

Comment:
I think this needs to be `.once`, since the listener should only fire once right. Otherwise we risk leaking memory by adding listeners but not removing them each time `transportOpen` is called

Reply:
Good point! 🙏